mpu9250_node:
  ros__parameters:
    freq: 100.0        # Update rate of the heading estimation and data published (at least lower than the chosen FIFO_SAMPLE_RATE)
    filter: 1                 # 0: NONE, 1: MADGWICK, 2: MAHONY

    # sensor specific parameters
    accel_fs_sel: 1           # Corresponds to ACCEL_FS_SEL::A4G
    gyro_fs_sel: 0            # Corresponds to GYRO_FS_SEL::G500DPS
    mag_output_bits: 1        # Corresponds to MAG_OUTPUT_BITS::M14BITS
    fifo_sample_rate: 1       # Corresponds to FIFO_SAMPLE_RATE::SMPL_500HZ
    gyro_dlpf_cfg: 2          # Corresponds to GYRO_DLPF_CFG::DLPF_41HZ
    accel_dlpf_cfg: 2         # Corresponds to ACCEL_DLPF_CFG::DLPF_45HZ
    g: 9.81

##### SENSOR SPECIFIC PARAMETERS ######
#                     0         1       2        3        4       5       6         7
# ACCEL_FS_SEL    {   2G,       4G,     8G,      16G      }
# GYRO_FS_SEL     {   250DPS,   500DPS, 1000DPS, 2000DPS  }
# MAG_OUTPUT_BITS {   14BITS,   16BITS  }
# FIFO_SAMPLE_RATE{   1000HZ,   500HZ,  333HZ,   250HZ,  200HZ,   167HZ,  143HZ,    125HZ, }
# GYRO_DLPF_CFG   {   250HZ,    184HZ,  92HZ,    41HZ,   20HZ,    10HZ,   5HZ,      3600HZ, }
# ACCEL_DLPF_CFG  {   218HZ_0,  218HZ_1, 99HZ,   45HZ,   21HZ,    10HZ,   5HZ,      420HZ, }
#
# more details at: https://invensense.tdk.com/wp-content/uploads/2015/02/PS-MPU-9250A-01-v1.1.pdf
